sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -97,6 +97,49 @@ kernel_pid_t gnrc_ipv6_init(void)
     return gnrc_ipv6_pid;
 }
 
+static void _dispatch_insert_ext_headers(gnrc_pktsnip_t *pkt)
+{
+    DEBUG("ipv6: dispatch insertion of extension headers\n");
+    uint8_t ext_demux[] = { PROTNUM_IPV6_EXT_HOPOPT
+                          , PROTNUM_IPV6_EXT_RH
+                          , PROTNUM_IPV6_EXT_FRAG
+                          , PROTNUM_IPV6_EXT_ESP
+                          , PROTNUM_IPV6_EXT_AH
+                          , PROTNUM_IPV6_EXT_DST
+                          , PROTNUM_IPV6_EXT_MOB
+                          };
+    bool encapsulate = false;
+
+    for ( size_t i = 0; i < sizeof(ext_demux); ++i ) {
+        gnrc_netreg_entry_t *call = gnrc_netreg_lookup(GNRC_NETAPI_MSG_TYPE_SND, ext_demux[i]);
```

Comment:
I don't like the fact that IPv6 needs to know something about, say RPL.
Additionally it would require some handshaking which is not needed by using IPC. 
> use raw IPC

I do, I iterate all interested PIDs with `gnrc_netreg_lookup()` and use plain message passing here.